### PR TITLE
Update one_hot() function 

### PR DESCRIPTION
**Pull request by myron**
Updates one_hot() function implementation. 

1. it's significantly faster on GPU (100x)
for (10,1, 256,256,256) input label shape, the time is 0.0002s vs 0.02s current version.
2. output is contiguous by construction, so no need to check if labels.is_contiguous()
3. it allows to specify the output type (e.g. float) instead of always having .long output type. 

### Status: **Ready**

I kept the old function as one_hot_v2()

PS: passes the unit tests in tests/test_to_onehot.py

### CODE REVIEW

File: monai/networks/utils.py [hunk 1]
```diff
@@ -17,8 +17,32 @@
...
+    """
+    For a tensor `labels` of dimensions B1[spatial_dims], return a tensor of dimensions `BN[spatial_dims]`
+    for `num_classes` N number of classes.
+
+    Example:
+
+        For every value v = labels[b,1,h,w], the value in the result at [b,v,h,w] will be 1 and all others 0.
+        Note that this will include the background label, thus a binary mask should be treated as having 2 classes.
+    """
+    assert labels.dim() > 1, "labels should have dim of 2 or more."
```
Comment (Nic-Ma) [inline monai/networks/utils.py]: This `assert` will cause the classification task in CI to fail, because the label shape can be [B] without channel.
Maybe you can `unqueeze` the data if it only has batch dim?
Then check:
```py
if labels.shape[1] != 1:
    raise ValueError("labels should have a channel with length equals to one.")
```

File: monai/networks/utils.py [hunk 2]
```diff
@@ -17,8 +17,32 @@
...
+
+    sh = list(labels.shape)
+    assert sh[1] == 1, "labels should have a channel with length equals to one."
+
+    sh[1] = num_classes
+    o = torch.zeros(size=sh, dtype=dtype, device=labels.device)
+    labels = o.scatter_(dim=1, index=labels.long(), value=1)
+
+    ## it's contiguous by construction, no need to check
```
Comment (Nic-Ma) [inline monai/networks/utils.py]: Please remove the comments if no use anymore.

File: monai/networks/utils.py [hunk 3]
```diff
@@ -17,8 +17,32 @@
...
+    # if not labels.is_contiguous():
+    #     labels = labels.contiguous()
+
+    return labels
 
-def one_hot(labels: torch.Tensor, num_classes):
+def one_hot_v2(labels: torch.Tensor, num_classes):
```
Comment (Nic-Ma) [inline monai/networks/utils.py:52]: You can remove this old version in your PR, so when your code is good enough, we can just replace it.
Thanks.

**Comment by Nic-Ma** [comment]
And could you please help add 2 unit tests to `test_to_onehot`?
1. Test the input with only `batch` dim.
2. Test the output match specified data type.

Thanks.

File: monai/networks/utils.py (hunk 1 above)
Comment (myron) [reply monai/networks/utils.py]: yes, I've noticed it too. it's fixed now

File: monai/networks/utils.py (hunk 3 above)
Comment (myron) [reply monai/networks/utils.py:52]: hmm, you sure? may be we can keep the old version for a bit just in case

Comment (Nic-Ma) [reply monai/networks/utils.py:52]: No need to keep 2 versions if no functional difference, looks confusing for users.
Thanks.

File: monai/networks/utils.py (hunk 2 above)
Comment (myron) [reply monai/networks/utils.py]: done

File: monai/networks/utils.py (hunk 3 above)
Comment (myron) [reply monai/networks/utils.py]: done

**Comment by myron** [comment]
all suggestions are included

**Comment by rabinadk1** [comment]
How about using the `torch.nn.functional.one_hot`?

**Comment by rabinadk1** [comment]
I haven't been able to pass the test cases but the rough code is in #4579
